### PR TITLE
Add with credential in search requests

### LINKED ISSUE #509: [Full-text search] Add withCredential in the query that requires autentication

### PR DESCRIPTION
**Pull request by sbrunner**
fix #509 

How to make it configurable?

### CODE REVIEW

File: src/services/creategeojsonbloodhound.js [hunk 1]
```diff
@@ -52,18 +53,28 @@ ngeo.CreateGeoJSONBloodhound;
  * @param {ol.proj.Projection=} opt_dataProjection Data projection.
  * @param {BloodhoundOptions=} opt_options optional Bloodhound options. If
  *     undefined, the default Bloodhound config will be used.
+ * @param {BloodhoundQueryOptions=} opt_query optional Bloodhound query options.
  * @return {Bloodhound} The Bloodhound object.
  */
 ngeo.createGeoJSONBloodhound = function(url, opt_filter, opt_featureProjection,
-    opt_dataProjection, opt_options) {
+    opt_dataProjection, opt_options, opt_query) {
```
Comment (ger-benjamin) [inline src/services/creategeojsonbloodhound.js]: If you set `opt_query` and `opt_options` with  `{remote: {...} }` then opt_query will be ignored.
That can be complicated to understand. can you add a note about that in the documentation ?

**Comment by ger-benjamin** [comment]
User can't set queryOptions. You need to add ``queryOptions: datasource.queryOptions`` in the cnfig object passed to the function "this.createDataset_" at line 196 in directive search.js

File: src/services/creategeojsonbloodhound.js [hunk 2]
```diff
@@ -52,18 +53,29 @@ ngeo.CreateGeoJSONBloodhound;
  * @param {ol.proj.Projection=} opt_dataProjection Data projection.
  * @param {BloodhoundOptions=} opt_options optional Bloodhound options. If
  *     undefined, the default Bloodhound config will be used.
+ * @param {BloodhoundQueryOptions=} opt_query optional Bloodhound query options.
+ *     Effective only it `remote`is not defined in `opt_options`.
```
Comment (ger-benjamin) [inline src/services/creategeojsonbloodhound.js]: Effective only it ... ->... if ...

**Comment by ger-benjamin** [comment]
Correct my last comment the merge. Thanks :-)

**Comment by sbrunner** [comment]
Thanks :-)

**Pull request merged**